src/o365/spo/commands/folder/folder-move.ts

```diff
@@ -0,0 +1,340 @@
+  }
+
+  /**
+   * Combines base and relative url 
+   * considering any missing slashes
+   * @param baseUrl https://contoso.com
+   * @param relativeUrl sites/abc
+   */
+  private urlCombine(baseUrl: string, relativeUrl: string): string {
+    // remove last '/' of base if exists
+    if (baseUrl.lastIndexOf('/') === baseUrl.length - 1) {
+      baseUrl = baseUrl.substring(0, baseUrl.length - 1);
+    }
+
+    // remove '/' at 0
+    if (relativeUrl.charAt(0) === '/') {
+      relativeUrl = relativeUrl.substring(1, relativeUrl.length);
+    }
+
+    // remove last '/' of next if exists
+    if (relativeUrl.lastIndexOf('/') === relativeUrl.length - 1) {
+      relativeUrl = relativeUrl.substring(0, relativeUrl.length - 1);
+    }
+
+    return `${baseUrl}/${relativeUrl}`;
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-u, --webUrl <webUrl>',
+        description: 'The URL of the site where the folder is located'
+      },
+      {
+        option: '-s, --sourceUrl <sourceUrl>',
+        description: 'Site-relative URL of the folder to move'
+      },
+      {
+        option: '-t, --targetUrl <targetUrl>',
+        description: 'Server-relative URL where to move the folder'
+      },
+      {
+        option: '--ignoreVersionHistory',
+        description: 'Ignores version history in the folder and will only move the main version'
+      },
+      {
+        option: '--allowSchemaMismatch',
+        description: 'Ignores any missing fields in the target and moves folder'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+
+      if (!args.options.webUrl) {
+        return 'Required parameter webUrl missing';
+      }
+
+      const isValidSharePointUrl: boolean | string = SpoCommand.isValidSharePointUrl(args.options.webUrl);
+      if (isValidSharePointUrl !== true) {
+        return isValidSharePointUrl;
+      }
+
+      if (!args.options.sourceUrl) {
+        return 'Required parameter sourceUrl missing';
+      }
+
+      if (!args.options.targetUrl) {
+        return 'Required parameter targetUrl missing';
+      }
+
+      return true;
+    };
+  }
+
+  public commandHelp(args: {}, log: (help: string) => void): void {
+    const chalk = vorpal.chalk;
+    log(vorpal.find(this.name).helpInformation());
+    log(
+      `  ${chalk.yellow('Important:')} before using this command, log in to a SharePoint Online site,
+    using the ${chalk.blue(commands.LOGIN)} command.
+  
+  Remarks:
+  
+    To move a folder, you have to first log in to SharePoint using the
+    ${chalk.blue(commands.LOGIN)} command, eg. ${chalk.grey(`${config.delimiter} ${commands.LOGIN} https://contoso.sharepoint.com`)}.
+
+    When you move a folder with documents that have version history,
```

Comment:
If this is always the case, why have the `ignoreVersionHistory` option?

Reply:
I have answered you in the other PR. I think we should remove `IgnoreVersionHistory`.

I guess getting to know the CLI and doing this has made me slightly confused - not seeing the forest for all the trees, will do better!